Hashdeep: add package

trldp opened the pull request:
Signed-off-by: Tobe Deprez <trldp@outlook.com>

Maintainer: trldp <trldp@outlook.com>
Compile tested: x86/64 on Ubuntu
Run tested: Linksys WRT1200AC (arm_cortex-a9_vfpv3), Openwrt 18.06

Description:
Add package for hashdeep. Hashdeep is a set of tools to compute and check MD5, SHA1, SHA256, tiger
and whirlpool hashsums of arbitrary number of files recursively.

I'm not sure I've entered the license correctly. The main license is US Govenment pulbic domain with some parts GPL-2. I have selected No Limit Public License (NLPL).

diizzyy commented:
md5 and sha256 is supported out of the box, sha1sum via coreutils
Wouldn't it make more sense to have a script doing this unless you need tiger and whirlpool algorithms?

trldp commented:
Hashdeep actually has more functionality than only computing and checking hashes recursively. For instance it can detect files that have been moved (i.e. have the same checksum, but different paths), detect files with hashes in a list of known hashes, ... None of this is impossible (or even hard) with a script of course, but it would take some time to implement.

diizzyy commented on utils/hashdeep/Makefile:
```diff
@@ -0,0 +1,48 @@
+  CATEGORY:=Utilities
+  TITLE:=Recursively compute hashsums or piecewise hashings
+  URL:=http://md5deep.sourceforge.net/
+  MAINTAINER:=Trldp <trldp@outlook.com>
```
Use your real name

diizzyy commented on utils/hashdeep/Makefile:
```diff
@@ -0,0 +1,48 @@
+
+define Package/hashdeep/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/hashdeep \
```
Use $(INSTALL_BIN) instead

trldp commented:
I've addressed your comments in a new commit

ja-pa commented on utils/hashdeep/Makefile:
```diff
@@ -6,8 +6,8 @@ PKG_RELEASE:=1
 PKG_LICENCE:=NLPL
 PKG_LICENCE_FILE:=COPYING
```
variable is PKG_LICENSE_FILES

trldp commented:
Whoops, sorry. Corrected now.

BKPepe commented:
@trldp please look again at variable PKG_LICENSE_FILES, create a new row between PKG_RELEASE and PKG_LICENSE and above PKG_LICENSE put there PKG_MAINTAINER and remove it from define Package/hashdeep. Then squash your commits.

neheb commented on utils/hashdeep/Makefile:
```diff
@@ -19,7 +19,7 @@ define Package/hashdeep
   URL:=http://md5deep.sourceforge.net/
-  MAINTAINER:=Trldp <trldp@outlook.com>
+  MAINTAINER:=Tobe Deprez <trldp@outlook.com>
   DEPENDS:=+libstdcpp
```
Does this require libstdcpp or can uclibc++ be used?

neheb replied:
Tested. Compiles just fine with uclibc++. Please change to $(CXX_DEPENDS) and include $(INCLUDE_DIR)/uclibc++.mk before package.mk

diizzyy replied:
If you want to do this, please run-test it first and preferably compare performance before blindly adding it.

BKPepe commented:
ping @trldp

trldp commented:
Sorry for the late reply, I was quite busy the last weeks. I've squashed everything into one commit included the remarks of @BKPepe. I've also switched to uclibc++ as @neheb suggested. I tested it on lede 17.01 on Linksys WRT1200AC (which is the only system I have available at the moment). Both the uclibc++ and stdcpp run fine with very similar performance.

diizzyy commented on utils/hashdeep/Makefile:
```diff
@@ -0,0 +1,49 @@
+endef
+
+define Build/Configure
+	(cd $(PKG_BUILD_DIR); sh bootstrap.sh; )
```
Doesnt PKG_FIXUP fix these issues?

trldp replied:
They do indeed. Fixed.
